$OpenBSD$

Restructure TCP keepalive constants for OpenBSD support.
GHC 9.10 CApiFFI can't resolve netinet/tcp.h macros on OpenBSD.
Create proper OpenBSD/Darwin/else branches, each with all 3 constants.

--- src/Simplex/Messaging/Transport/KeepAlive.hs.orig
+++ src/Simplex/Messaging/Transport/KeepAlive.hs
@@ -50,14 +50,22 @@
 #else
 -- Mac/Linux

-#if defined(darwin_HOST_OS)
+#if defined(openbsd_HOST_OS)
+-- GHC 9.10 CApiFFI can't resolve netinet/tcp.h macros
+_TCP_KEEPIDLE :: CInt
+_TCP_KEEPIDLE = 0x100
+_TCP_KEEPINTVL :: CInt
+_TCP_KEEPINTVL = 0x200
+_TCP_KEEPCNT :: CInt
+_TCP_KEEPCNT = 0x300
+#elif defined(darwin_HOST_OS)
 foreign import capi "netinet/tcp.h value TCP_KEEPALIVE" _TCP_KEEPIDLE :: CInt
+foreign import capi "netinet/tcp.h value TCP_KEEPINTVL" _TCP_KEEPINTVL :: CInt
+foreign import capi "netinet/tcp.h value TCP_KEEPCNT" _TCP_KEEPCNT :: CInt
 #else
 foreign import capi "netinet/tcp.h value TCP_KEEPIDLE" _TCP_KEEPIDLE :: CInt
-#endif
-
 foreign import capi "netinet/tcp.h value TCP_KEEPINTVL" _TCP_KEEPINTVL :: CInt
-
 foreign import capi "netinet/tcp.h value TCP_KEEPCNT" _TCP_KEEPCNT :: CInt
+#endif

 #endif
